### PR TITLE
Introduces log output class

### PR DESCRIPTION
**Pull request by ebozduman**
Fix for issue #574

### CODE REVIEW

File: tests/functional/tests.py [hunk 1]
```diff
@@ -75,35 +72,96 @@ def read(self, amt=64*1024):
...
-            'alert': alert,\
-            'message': None,\
-            'error': None,\
-            'status': PASS}
-
-def print_json(log_output):
-    print(json.dumps(log_output))
+class LogOutput(object):
+    """
+    LogOutput isi the class for log output. It is requried standard for all
```
Comment (nitisht) [inline tests/functional/tests.py]: LogOutput _is_ the class for log output. It is _required_ for all

**Comment by nitisht** [comment]
logging looks fine 

```
{"name": "minio-py", "args": {"bucket_name": "a3084782-4267-47a5-adfd-e586b5b72bfd"}, "duration": 2.31, "status": "PASS", "function": "make_bucket(bucket_name, location)", "description": "Tests make_bucket api"}
{"name": "minio-py", "args": {}, "duration": 1.84, "status": "PASS", "function": "list_buckets()", "description": "Tests list_buckets api"}
{"name": "minio-py", "args": {"metadata": {"x-amz-storage-class": "STANDARD_IA"}, "file_path": "datafile-1-MB", "object_name": "141d843a-d6e0-4f45-9bdf-b358aaffa703", "bucket_name": "1f47e701-dcd6-43c7-be5d-01d55cf7f824"}, "duration": 3.88, "status": "PASS", "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata)", "description": "Tests fput_object api"}
{"name": "minio-py", "args": {"metadata": {"x-amz-storage-class": "STANDARD_IA"}, "file_path": "datafile-11-MB", "object_name": "3a77f8e2-7aaa-4ef7-9a3a-e7a93208866e", "bucket_name": "7903ecd7-6e4a-47fd-b986-2c552c9c6fc5"}, "duration": 21.2, "status": "PASS", "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata)", "description": "Tests fput_object api"}
{"name": "minio-py", "args": {"object_name": "a774a1e1-74d9-4744-a4f7-a45fbf3536ee-copy", "conditions": {"set_match_etag": "test-etag"}, "object_source": "a774a1e1-74d9-4744-a4f7-a45fbf3536ee-source", "bucket_name": "c14add9d-b270-486f-bc47-a3a9fccf816c"}, "duration": 2.76, "status": "PASS", "function": "copy_object(bucket_name, object_name, object_source, conditions)", "description": "Tests copy_object api"}
{"name": "minio-py", "args": {"data": "LimitedRandomReader(MB_11)", "object_name": "2b5f564d-81c8-4dc2-bbfa-2e01e6a51012", "length": 11534336, "bucket_name": "4047ac5f-d83b-4396-a161-0072982e0c44", "metadata": {"x-amz-meta-testing": "value", "Content-Type": "application/octet-stream"}}, "duration": 25.09, "status": "PASS", "function": "put_object(bucket_name, object_name, data, length, content_type, metadata)", "description": "Tests put_object api"}
{"name": "minio-py", "args": {"object_name": "aa3eb2cb-decc-4e37-a27a-1dedbaea8f50", "bucket_name": "beff9633-ab45-4180-9263-783aba593a2b"}, "duration": 5.73, "status": "PASS", "function": "get_object(bucket_name, object_name, request_headers)", "description": "Tests get_object api"}
{"name": "minio-py", "args": {"file_path": "newfile-f \u65b0", "object_name": "35213b86-0b4f-45ab-a41e-0a12f55c603b", "bucket_name": "1c3a4e0a-ecf4-4374-98f1-6b3abe1794c1"}, "duration": 10.96, "status": "PASS", "function": "fget_object(bucket_name, object_name, file_path, request_headers)", "description": "Tests fget_object api"}
{"name": "minio-py", "args": {"object_name": "659455b0-3d2b-4e63-b0ad-e78fd7ac1add", "recursive": true, "bucket_name": "6bbacd6b-446e-498c-80a6-455a13220053"}, "duration": 5.26, "status": "PASS", "function": "list_objects(bucket_name, prefix, recursive)", "description": "Tests list_objects api"}
{"name": "minio-py", "args": {"object_name": "9fd22dde-155f-40d5-bb71-23dc9df90d2e", "recursive": true, "bucket_name": "da10fe4f-0640-44e4-a541-c0754eff93f6"}, "duration": 3.55, "status": "PASS", "function": "list_objects_v2(bucket_name, prefix, recursive)", "description": "Tests list_objects_v2 api"}
{"name": "minio-py", "args": {"object_name": "ab327471-eeaf-40c6-93cc-c15f9a4d8587", "bucket_name": "6a0eeeae-d1bc-48b4-a22d-41d087e2d480"}, "duration": 11.06, "status": "PASS", "function": "presigned_get_object(bucket_name, object_name, expires, response_headers)", "description": "Tests presigned_get_object api"}
{"name": "minio-py", "args": {"object_name": "cd1efe3f-b96b-4411-ad29-c30d21b67307", "bucket_name": "67cf7d36-0d4e-4753-8f7b-a8eac9852b9d"}, "duration": 2.56, "status": "PASS", "function": "presigned_put_object(bucket_name, object_name, expires)", "description": "Tests presigned_put_object api"}
{"name": "minio-py", "args": {"post_policy": {"prefix": "objectPrefix/", "expires_in_days": 10, "bucket_name": "6feebbbc-1c2a-4286-bdb7-04e2513ac925"}}, "duration": 0.61, "status": "PASS", "function": "presigned_post_policy(post_policy)", "description": "Tests presigned_post_policy api"}
{"name": "minio-py", "args": {"bucket_name": "a53063f8-2db5-4b44-a4c5-46432fe484ac"}, "duration": 1.13, "status": "PASS", "function": "get_bucket_policy(bucket_name, prefix)", "description": "Tests get_bucket_policy api"}
{"name": "minio-py", "args": {"prefix": "1/", "prefix-2": "", "bucket_name": "0c145652-6d8d-48c9-a91a-05a07ede24d1"}, "duration": 2.87, "status": "PASS", "function": "set_bucket_policy(bucket_name, prefix, policy_access)", "description": "Tests set_bucket_policy api"}
{"name": "minio-py", "args": {"object_name": "4ab9b243-c412-455d-8323-19dc0f8a2741", "bucket_name": "f6ea5c17-87e8-4b88-a59e-90ffe30781ee"}, "duration": 1.13, "status": "PASS", "function": "remove_object(bucket_name, object_name)", "description": "Tests remove_object api"}
{"name": "minio-py", "args": {"objects_iter": ["prefix-0", "prefix-1", "prefix-2", "prefix-3", "prefix-4", "prefix-5", "prefix-6", "prefix-7", "prefix-8", "prefix-9"], "bucket_name": "98cf3889-873d-42bd-802e-e12609d0129a"}, "duration": 12.7, "status": "PASS", "function": "remove_objects(bucket_name, objects_iter)", "description": "Tests remove_objects api"}
{"name": "minio-py", "args": {"bucket_name": "35808400-8b8b-44e7-848c-04cb9d1f005e"}, "duration": 0.61, "status": "PASS", "function": "remove_bucket(bucket_name)", "description": "Tests remove_bucket api"}
```

**Comment by vadmeste** [comment]
@ebozduman, we usually follow this style when submitting a new PR:
- Only one commit: so here you need to squash your two commits into one.
- "Fixes #ISSUE_ID" is usually written in the PR body and not in the title. Github will automatically add the PR_ID to the commit title. Then, if someone wants to inform himself about the whole story, he will be able to open the merged pull request and find the issue id.

File: tests/functional/tests.py [hunk 2]
```diff
@@ -114,16 +172,12 @@ def test_make_bucket(client):
         # Remove bucket
         client.remove_bucket(bucket_name)
     except Exception as err:
-        log_output['message'] = err
-        log_output['error'] = traceback.format_exc()
-        log_output['status'] = FAIL
-        log_output['duration'] = duration = round(time.time() - start_time, 2)
-        print_json(log_output)
+        print(log_output.json_report_failure(err))
```
Comment (vadmeste) [inline tests/functional/tests.py]: Good change,

But I still want to to recommend that you only throw the exception here, the main function which is calling all test functions will know that an error occurred and will print it. I think this style is better than `print() > exit()` in each test function.

**Pull request merged**